src/services/exposed/records-getter.js

```diff
@@ -23,67 +26,82 @@ class RecordsGetter extends AbstractRecordService {
     return records;
   }
 
-  // NOTICE: This function accept either query or ID list params and return an ID list.
-  //         It could be used to handle both "select all" (query) and "select some" (ids).
-  //         It also handles related data.
-  async getIdsFromRequest(params) {
-    const isRelatedData = (attributes) =>
-      attributes.parentCollectionId
-      && attributes.parentCollectionName
-      && attributes.parentAssociationName;
-
-    const recordsGetter = async (attributes) => {
-      const { parentCollectionId, parentCollectionName, parentAssociationName } = attributes;
-      if (isRelatedData(attributes)) {
-        const parentModel = this.modelsManager.getModels()[parentCollectionName];
-        const [records] = await new this.Implementation.HasManyGetter(
-          parentModel,
-          this.model,
-          this.lianaOptions,
-          {
-            ...this.params,
-            ...params,
-            ...attributes.allRecordsSubsetQuery,
-            page: attributes.page,
-            recordId: parentCollectionId,
-            associationName: parentAssociationName,
-          },
-          this.user,
-        ).perform();
-        return records;
-      }
-      return this.getAll({ ...attributes.allRecordsSubsetQuery, page: attributes.page });
-    };
+  /**
+   * Takes a request from a frontend bulk action (bulk smart-action, bulk delete, ...) and
+   * returns the list of ids that should be affected.
+   *
+   * @fixme why are we testing for attrs?.allRecords !== true && attrs.ids to detect "all records"
+   *        queries? IMHO this should be only attrs?.allRecords
+   *
+   * @fixme Composite ids are returned separated by a dash "-".
+   *        Not sure why, we are using pipes "|" in forest-express-sequelize but changing it
+   *        would be a breaking change.
+   */
+  async getIdsFromRequest(request) {
+    const attrs = new QueryDeserializer(request?.body?.data?.attributes ?? {}).perform();
+    const idsExcludedAsString = attrs?.allRecordsIdsExcluded?.map?.(String) ?? [];
+
+    // "select all records" is not selected
+    if (attrs?.allRecords !== true && attrs.ids) {
+      return attrs.ids;
+    }
+
+    // Otherwise, query database in a loop to retrieve all ids.
+    const ids = [];
+    const pageSize = 1000;
```

Comment:
Actually it's missing one small improvement is here. The end user should be able to configure the paging. 
```suggestion
    const pageSize = process.env.FOREST_IDS_PAGE_SIZE || 5000;
```
1000 is way too small I think by default